plugins/out_kinesis_streams/kinesis_api.c

```diff
@@ -97,6 +97,61 @@ static int init_put_payload(struct flb_kinesis *ctx, struct flush *buf,
     return -1;
 }
 
+/*
+ * Count the substring number in string.
+ */
+static int get_substring_number(char *str1, char *str2){
+    int x = 0;
+    char *p;
+
+        do{
+        p = strstr(str1, str2);
+        if (p != NULL) {
+            str1 = p + 1;
+            x = x + 1;
+        }
+    }
+    while(p != NULL);
+    return x;
+}
+
+/*
+ * Adjust the time_key output format according to time_key_format.
+ */
+static void get_time_format_string(char *str, size_t maxsize,
+                      const char *time_key_format, const struct tm *timeptr,
+                      size_t len, struct flb_time *tms){
+    int millisecond_length = ((int)((ceil(log10(abs((uint64_t) tms->tm.tv_nsec / 1000000))) + 1)*sizeof(char)));
+    int nanosecond_length = ((int)((ceil(log10(abs((uint64_t) tms->tm.tv_nsec)))+1)*sizeof(char)));
+    char millisecond_string[millisecond_length];
+    char nanosecond_string[nanosecond_length];
+    char *milli_string_old = "%F";
+    char *nano_string_old = "%L";
```

Comment:
I do not understand these variable names and also I think these should be constants with #define instead of vars